src/Microsoft.ML.Core/Data/RoleMappedSchema.cs

```diff
@@ -98,68 +112,99 @@ public sealed class RoleMappedSchema
         private const string NameString = "Name";
         private const string FeatureContributionsString = "FeatureContributions";
 
+        /// <summary>
+        /// Instances of this are the keys of a <see cref="RoleMappedSchema"/>. This class also some important commonly used
+        /// pre-defined instances available (e.g., <see cref="Label"/>, <see cref="Feature"/>) that should be used when
+        /// possible for consistency reasons. However, practitioners should not be afraid to declare custom roles if
+        /// approppriate for their task.
+        /// </summary>
         public struct ColumnRole
         {
+            /// <summary>
+            /// Role for features. Commonly used as the independent variables given to trainers, and scorers.
+            /// </summary>
             public static ColumnRole Feature => FeatureString;
+
+            /// <summary>
+            /// Role for labels. Commonly used as the dependent variables given to trainers, and evaluators.
+            /// </summary>
             public static ColumnRole Label => LabelString;
+
+            /// <summary>
+            /// Role for group ID. Commonly used in ranking applications, for defining query boundaries, or
+            /// sequence classification, for defining the boundaries of an utterance.
+            /// </summary>
             public static ColumnRole Group => GroupString;
+
+            /// <summary>
+            /// Role for sample weights. Commonly used to point to a number to make trainers give more weight
+            /// to a particular example.
+            /// </summary>
             public static ColumnRole Weight => WeightString;
             public static ColumnRole Name => NameString;
```

Comment:
>public static ColumnRole Name => NameString; [](start = 12, length = 44)

considering you make reference to this column (see line 205) it's probably make sense to add description for this as well.

Reply:
Yes... also I think FeatureContributions should be done away with, I'm not sure why it is here. Maybe someone thought once upon a time that for a column role to be used, it has to be here.

---
In reply to: [200438589](https://github.com/dotnet/machinelearning/pull/496#discussion_r200438589) [](ancestors = 200438589)